src/test/objectstore/allocsim/ops_scraper.py

```diff
@@ -0,0 +1,290 @@
+
+def process_osd(osd_id) -> ProcessInfo:
+    if osd_id not in Env.store():
+        Env.store()[osd_id] = {}
+        Env.store()[osd_id]['last_op'] = datetime.datetime.fromtimestamp(0, tz=datetime.timezone.utc)
+
+
+    logger.info(f'collecting from osd.{osd_id}')
+    command_time_start = time.time()
+    historic: Dict[str, Any] = run_ceph_command(['tell', f'osd.{osd_id}', 'dump_historic_ops'])
+    command_time = time.time() - command_time_start
+
+    osd_processing_start = time.time()
+
+    operations_str = 'op_type | offset/extent | name | pgid | initiated_at | who\n'
+    operations_str += '-------------------------------------------------------\n'
+    new_ops = 0 
+    oldest_operation = None
+    capture_period_start = _to_timestamp(Env.store()[osd_id]['last_op'])
+    for op in historic['ops']:
+        initiated_at = datetime.datetime.fromisoformat(op['initiated_at'])
+        if initiated_at < Env.store()[osd_id]['last_op']:
+            continue
+
+        Env.store()[osd_id]['last_op'] = initiated_at
+        description = op['description']
+        logger.debug(f'{description}')
+        if description.startswith('osd_op('):
+            new_ops += 1
+            description_data = description[7:-1]
+            op = parse_osd_op(description_data, initiated_at)
+            if not oldest_operation:
+                oldest_operation = op
+            operations_str += f'{str(op)}\n'
+    capture_period_end = time.time()
+    processing_time = time.time() - osd_processing_start
+    logger.info(f'osd.{osd_id} new_ops {new_ops}')
+
+    return ProcessInfo(processing_time, command_time, operations_str, len(historic['ops']), 
+                       new_ops, oldest_operation, capture_period_start, capture_period_end)
+
+def _set_history_size(history_size: int):
+    run_ceph_command(['config', 'set', 'osd', 'osd_op_history_size', 
+                      str(int(history_size))], no_out=True)
+
+def main():
+    parser = argparse.ArgumentParser(
+            prog='OSD operations parser')
+    parser.add_argument('--nohash',
+                        action='store_true', required=False)
+    parser.add_argument('--debug_level', type=str, default='warning')
+    parser.add_argument('--ceph_bin_path', required=False)
+    parser.add_argument('--freq', required=False, type=int, default=1)
+    parser.add_argument('--min_history_size', required=False, type=int, default=100)
+    parser.add_argument('--max_history_size', required=False, type=int, default=1000)
+    parser.add_argument('--out', required=False, help="filename to write output to. If none is provided it will be written to stdout")
+    args = parser.parse_args()
+
+    Env.setup_env(args)
+
+    log_levels = {
+            'CRITICAL': logging.CRITICAL,
+            'ERROR': logging.ERROR,
+            'WARNING': logging.WARNING,
+            'INFO': logging.INFO,
+            'DEBUG': logging.DEBUG,
+            'NOTSET': logging.NOTSET
+    }
```

Comment:
Names for debug levels are good, but in output.
I would imagine that one would prefer to do "-v -v -v -v" or "--debug=4" rather then check the naming "--debug=info" and then changing "--debug=debug" when that turned to be not enough.